core/tests/puppeteer-acceptance-tests/spec/blog-editor-tests/try-to-publish-a-duplicate-blog-post-and-get-blocked.spec.ts

```diff
@@ -16,21 +16,22 @@
  * @fileoverview Acceptance Test for Blog Post Editor
  */
 
-const userFactory = require(
-  '../../puppeteer-testing-utilities/user-factory.js');
-const testConstants = require(
-  '../../puppeteer-testing-utilities/test-constants.js');
+import * as userFactory from
+  '../../puppeteer-testing-utilities/user-factory';
+import testConstants from
+  '../../puppeteer-testing-utilities/test-constants';
+import { IBlogPostEditor } from '../../user-utilities/blog-post-editor-utils';
 
 const DEFAULT_SPEC_TIMEOUT = testConstants.DEFAULT_SPEC_TIMEOUT;
 const duplicateBlogPostWarning = ' Blog Post with the' +
   ' given title exists already. Please use a different title. ';
 
 describe('Blog Editor', function() {
-  let blogPostEditor = null;
+  let blogPostEditor: IBlogPostEditor;
 
   beforeAll(async function() {
-    blogPostEditor = await userFactory.createNewBlogPostEditor(
-      'blogPostEditor');
+    blogPostEditor = await userFactory.createNewUser(
+      'blogPostEditor', 'blog_post_editor@example.com', ['blog post editor']);
```

Comment:
We should probably make things like 'blog post editor' named constants rather than raw strings.

Reply:
These strings are typed since they are keys of the mapping corresponding to the classes so typescript will complain if you enter some random string. However, they can also be put in test-constants WDYT.

Reply:
Oh that's good ... but not apparent. Hm. I think I prefer them as constants, so that devs get used to the idea of not using raw strings directly in code.

Reply:
Done, added a new roles object in test constants.